### DISCUSSION TITLE
Have an option to switch between the built in tablet driver and an outside one.

### DISCUSSION THREAD

**Discussion by Tresquel** [ppy/osu#15580] [Ideas]
My tablet has *a lot* of smoothing and I have opentabletdriver installed and configured how I'd want it to but when I switch to osu it switches to it's own driver which results in a ton of smoothing and disabling the tablet option inside osu disables it in the osu window completely.

It happens for me on Linux I don't know if it happens on Windows or Mac too.

**Comment by bdach** [comment]
you mean like this one?

![2021-11-11-205802_474x98_scrot](https://user-images.githubusercontent.com/20418176/141360909-d16e9c56-0769-4d4b-9586-e12653c4d040.png)

**Reply by Tresquel** [reply]
> you mean like this one?
> 
> ![2021-11-11-205802_474x98_scrot](https://user-images.githubusercontent.com/20418176/141360909-d16e9c56-0769-4d4b-9586-e12653c4d040.png)

No, since that switches off the tablet from working with osu completely (not turning it off just osu seems to ignore it's inputs).

**Reply by bdach** [reply]
that really shouldn't be happening. try switching off "high precision mouse" too, if you have that enabled.

**Reply by Raidoki** [reply]
> > you mean like this one?
> > ![2021-11-11-205802_474x98_scrot](https://user-images.githubusercontent.com/20418176/141360909-d16e9c56-0769-4d4b-9586-e12653c4d040.png)
> 
> No, since that switches off the tablet from working with osu completely (not turning it off just osu seems to ignore it's inputs).

Hey, i have the same issue but in a different case because my 32:9 is not supported correctly. But if i turn of the Tablet ingame i get no inputs too :/

**Reply by Tresquel** [reply]
> that really shouldn't be happening. try switching off "high precision mouse" too, if you have that enabled.

That option doesn't do anything, game still favours the in-game driver.

**Reply by bdach** [reply]
i'm 99% sure linux will handle tablets like pointing devices/mice. if it doesn't in your case, then i'm almost certain it's because of either your device or your system configuration.

**Reply by Tresquel** [reply]
I still don't think that this should be happening:

https://user-images.githubusercontent.com/43729305/141530895-24c1e328-26a6-4e09-9352-94ff4395d4a4.mp4

**Reply by Tresquel** [reply]
Alright so all this mess is caused mostly by me being dumb and partly by osu using something else for millimeters as measurements for the tablet area. The video above clearly had two different tablet areas selected in osu and OTD and I thought it could help visualize the problem while only making it worse. 
Now when I do set my tablet area to the same values as in open tablet driver it feels bigger and more sluggish than normal and while switching in fullscreen to another window it does seem to have some cursor jump while having the same values as in OTD. Sorry for kinda wasting your time tho the tablet settings inside osu are pretty hard to use. Maybe make it update when changing the values (not just when confirming them but also while holding down the slider the area should move around) and make it use millimeters.

**Reply by Tresquel** [reply]
Just found another thing: if you move your cursor around while sliding a slider in options to make your cursor spin, it'll spin after letting go of the slider even tho it shouldn't (at least I think so)

**Reply by bdach** [reply]
> if you move your cursor around while sliding a slider in options to make your cursor spin, it'll spin after letting go of the slider even tho it shouldn't (at least I think so)

that is intentional.

**Reply by peppy** [reply]
You can turn off the cursor spinning if you'd prefer, it's in the options.
